tests/unittests/CoreGraphics/CGBitmapContextTests.mm

```diff
@@ -65,4 +65,105 @@
 
     CGColorSpaceRelease(grayColorSpace);
     CGContextRelease(context);
-}
\ No newline at end of file
+}
+
+void _DrawText(CGContextRef context, NSString* text, const CGRect bounds) {
+    // Aligns origin for our frame
+    CGContextTranslateCTM(context, 0.0f, bounds.size.height);
+
+    // Flips y-axis for our frame
+    CGContextScaleCTM(context, 1.0f, -1.0f);
+
+    // Creates path with current rectangle
+    CGMutablePathRef path = CGPathCreateMutable();
+    CGPathAddRect(path, NULL, bounds);
+
+    CTFontRef myCFFont = CTFontCreateWithName((__bridge CFStringRef) @"Arial", 10, NULL);
+    CFAutorelease(myCFFont);
+
+    // Make dictionary for attributed string with font, color, and alignment
+    NSDictionary* attributesDict = [NSDictionary dictionaryWithObjectsAndKeys:(__bridge id)myCFFont,
+                                                                              (id)kCTFontAttributeName,
+                                                                              kCFBooleanTrue,
+                                                                              (id)kCTForegroundColorFromContextAttributeName,
+                                                                              nil];
+
+    CFAttributedStringRef attrString =
+        CFAttributedStringCreate(kCFAllocatorDefault, (__bridge CFStringRef)text, (__bridge CFDictionaryRef)attributesDict);
+    CFAutorelease(attrString);
+
+    CTFramesetterRef framesetter = CTFramesetterCreateWithAttributedString(attrString);
+    CFAutorelease(framesetter);
+
+    // Creates frame for framesetter with current attributed string
+    CTFrameRef frame = CTFramesetterCreateFrame(framesetter, CFRangeMake(0, 0), path, NULL);
+    CFAutorelease(frame);
+
+    // Draws the text in the frame
+    CTFrameDraw(frame, context);
+}
+
+// we will create a 10x10 bitmap context and :
+// 1. Fill context with R=0.0, G=0.0, B=0.5, A=1.
+// data buffer in little endian format: R = 0, G=0, B > 0, A = FF
+// 2. Set R=1, G=0.5, B=0, A=1.0
+// Draw "||" into the buffer
+// The data buffer should contain R > G > B.
+// Due to text aliasing, we won't get the actual values we specify hence this
+
+// The test will break under two circumstances:
+// 1. pixel format breaks due to CG drawing code change (for as long as we use cairo)
+// 2. pixel format breaks due to changes in render target pixel format for bitmap context.
+
+void _TestPixelFormat(const CGRect bounds, const CGBitmapInfo info, DWORD expectedBG, DWORD expectedFG1, DWORD expectedFG2) {
+    CGColorSpaceRef colorSpace = CGColorSpaceCreateDeviceRGB();
+
+    const size_t width = bounds.size.width;
+    const size_t height = bounds.size.height;
+    const size_t bytesPerPixel = 4;
+    const size_t bitsPerChannel = 8;
+    const CGFloat color[] = { 1.0, 0.5, 0.0, 1.0 };
+    const CGFloat bgColor[] = { 0.0, 0.0, 0.5, 1.0 };
+
+    woc::unique_iw<uint8_t> data(static_cast<uint8_t*>(IwMalloc(width * height * 4)));
```

Comment:
why not std::vector here?  and will unique_ptr properly delete the malloced array when created only as a pointer?